src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3048 @@
+
+        env.fund(XRP(1000), alice, bob);
+        env.close();
+
+        std::uint32_t aliceTicketSeq{env.seq(alice) + 1};
+        env(ticket::create(alice, 10));
+        env.close();
+
+        auto const aliceSeq = env.seq(alice);
+        auto const preAlice = env.balance(alice);
+        auto const preBob = env.balance(bob);
+
+        auto const batchFee = batch::calcBatchFee(env, 0, 2);
+        env(batch::outer(alice, 0, batchFee, tfAllOrNothing),
+            batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+            batch::inner(pay(alice, bob, XRP(2)), aliceSeq),
+            ticket::use(aliceTicketSeq));
+        auto const txIDs = env.tx()->getBatchTransactionIDs();
+        TxID const parentBatchId = env.tx()->getTransactionID();
+        std::vector<TestBatchData> testCases = {
+            {"tesSUCCESS", to_string(txIDs[0])},
+            {"tesSUCCESS", to_string(txIDs[1])},
+        };
+        env.close();
+        validateBatch(env, parentBatchId, testCases);
+
+        auto const sle = env.le(keylet::account(alice));
+        BEAST_EXPECT(sle);
+        BEAST_EXPECT(sle->getFieldU32(sfOwnerCount) == 8);
+        BEAST_EXPECT(sle->getFieldU32(sfTicketCount) == 8);
+
+        BEAST_EXPECT(env.seq(alice) == aliceSeq + 1);
+        BEAST_EXPECT(env.balance(alice) == preAlice - XRP(3) - batchFee);
+        BEAST_EXPECT(env.balance(bob) == preBob + XRP(3));
+    }
+
+    void
+    testSequenceOpenLedger(FeatureBitset features)
+    {
+        testcase("sequence open ledger");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+
+        // Before Batch Txn w/ same sequence
+        // Batch inner using non existing ticket
+        {
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const batchTxn = env.jt(
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(
+                    pay(alice, bob, XRP(1)),
+                    0,
+                    aliceSeq + 1),  // Uses Ticket (Fails with terPRE_TICKET)
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            env(noopTxn, ter(terPRE_SEQ));
+            env(batchTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1,
+                 "AccountSet",
```

Comment:
where did the AccountSet tx come from? both inner txns are Payment

Reply:
This is testing the full ledger. The noop transaction is in the open ledger before the batch was submitted. Both inner transactions failed, because its an all or nothing and a `terPRE_TICKET` error

Reply:
would it be possible to add comments in the test to describe this behavior?